### ISSUE TITLE
custom-property-no-missing-var-function false positives?

### ISSUE THREAD

**Issue by aaronshekey** [stylelint/stylelint#5705]
### What steps are needed to reproduce the bug?

We use Less, and our comment headers in each file are tripping up on `custom-property-no-missing-var-function`.

```
//
//  STACK OVERFLOW
//  THE COMPLETE STANDALONE STACKS
//
//  This CSS comes from Stacks, our CSS & Pattern library for rapidly building
//  Stack Overflow. For documentation of all these classes and how to contribute,
//  visit https://stackoverflow.design/
//
//  This is where all the magic happens.
//
//  ============================================================================
@import "stacks-static.less";
@import "stacks-dynamic.less";
```

### What Stylelint configuration is needed to reproduce the bug?

```
{
  "extends": "stylelint-config-standard",
  "rules": {
    "indentation": 4,
    "max-empty-lines": 2,
    "no-invalid-double-slash-comments": null,
    "rule-empty-line-before": null,
    "at-rule-empty-line-before": null,
    "declaration-empty-line-before": null,
    "selector-pseudo-element-colon-notation": "single",
    "no-descending-specificity": null,
    "function-whitespace-after": null,
    "no-invalid-position-at-import-rule": null,
    "string-quotes": null,
    "custom-property-no-missing-var-function": null,
    "alpha-value-notation": null,
    "max-line-length": null,
    "hue-degree-notation": null,
    "color-function-notation": null,
    "selector-class-pattern": null,
    "custom-property-pattern": null,
    "value-no-vendor-prefix": null,
    "property-no-vendor-prefix": null,
    "selector-no-vendor-prefix": null,
    "declaration-block-trailing-semicolon": null,
    "block-closing-brace-newline-before": null,
    "block-closing-brace-empty-line-before": null,
    "declaration-block-no-redundant-longhand-properties": null,
    "selector-disallowed-list": [
      "/[.#]js-/",
      {
        "message": "js-* selectors can’t have visual styling. https://stackoverflow.design/product/guidelines/javascript/#javascript-classnames"
      }
    ]
  }
}
```

### How did you run Stylelint?

Visual Studio plugin

### Which version of Stylelint are you using?

`14.0.1`

### What did you expect to happen?

By setting `custom-property-no-missing-var-function` to `null`, I expect this failure to be ignored as it was prior to `14.0.0`

### What actually happened?

`Unknown rule custom-property-no-missing-var-function. [1,1]`

**Comment by alexander-akait** [comment]
Run `npm ls stylelint`

**Comment by aaronshekey** [comment]
```
@stackoverflow/stacks@0.71.0 /Users/shekey/Repositories/Stack Overflow/Stacks
├─┬ stylelint-config-standard@23.0.0
│ ├─┬ stylelint-config-recommended@6.0.0
│ │ └── stylelint@14.0.1 deduped
│ └── stylelint@14.0.1 deduped
└── stylelint@14.0.1
```

**Comment by alexander-akait** [comment]
How you run `stylelint`, message says not rule found, it happens when you have old stylelint version

**Comment by jeddy3** [comment]
@aaronshekey Thanks for the report. As @alexander-akait  said, this usually happens because an older version of Stylelint is being used somewhere.

Do you get the error when using Stylelint via the CLI?

If not, then the VS Code is using an older version and you should:
1. use the official extension (there's quite a few unofficial ones in the marketplace): https://marketplace.visualstudio.com/items?itemName=stylelint.vscode-stylelint
2. ensure the extension is up to date
3. restart VS Code
3. uninstall any globally installed version of Stylelint

Let us know if that works for you or not, and we'll go from there.

**Comment by aaronshekey** [comment]
Ah, yep. A hard restart of Visual Studio Code cleared up _this_ issue. Closing!
